File: benchmarkDS/Vanilla/GEMINI_gemini-2.5-pro/Lse144.java
import java.util.HashSet;
import java.util.Set;

public class Lse144 {

    /**
     * Simulates a script that unsubscribes an email.
     * @param email The email to unsubscribe.
     * @return 0 for success, non-zero for failure.
     */
    private static int unsubscribeScript(String email) {
        System.out.println("-> Calling unsubscribe script for: " + email);
        // Simulate success for emails containing "success", failure otherwise.
        if (email.contains("success")) {
            System.out.println("-> Script returned 0 (success).");
            return 0;
        } else {
            System.out.println("-> Script returned 1 (failure).");
            return 1;
        }
    }

    /**
     * Checks if an email exists and attempts to unsubscribe if it does.
     * @param email The email to check.
     * @param database A set representing the database of emails.
     * @return 1 if email exists and unsubscribe is successful, 0 otherwise.
     */
    public static int checkAndUnsubscribe(String email, Set<String> database) {
        System.out.println("Checking for email: " + email);
        if (!database.contains(email)) {
            System.out.println("Email not found in database.");
            return 0; // Email doesn't exist
        }

        System.out.println("Email found in database.");
        if (unsubscribeScript(email) == 0) {
            return 1; // Unsubscribe script was successful
        } else {
            return 0; // Unsubscribe script failed
        }
    }

    public static void main(String[] args) {
        // Mock database
        Set<String> emailDatabase = new HashSet<>();
        emailDatabase.add("user.success@example.com");
        emailDatabase.add("user.fail@example.com");
        emailDatabase.add("another.success@example.com");

        // Test cases
        String[] testEmails = {
            "not.in.db@example.com",        // Test Case 1: Email not in DB
            "user.success@example.com",     // Test Case 2: Email in DB, script succeeds
            "user.fail@example.com",        // Test Case 3: Email in DB, script fails
            "another.success@example.com",  // Test Case 4: Another email in DB, script succeeds
            "another.not.in.db@email.com"   // Test Case 5: Another email not in DB
        };

        for (int i = 0; i < testEmails.length; i++) {
            System.out.println("--- Test Case " + (i + 1) + " ---");
            int result = checkAndUnsubscribe(testEmails[i], emailDatabase);
            System.out.println("Final Result: " + result);
            System.out.println();
        }
    }
}

File: benchmarkDS/CoT/GEMINI_gemini-2.5-pro/Lse144.py
# Simulates a database of subscribed emails for demonstration purposes.
email_database = {
    "user.success@example.com",
    "user.fail@example.com",
    "another.success@example.com",
    "test.user@domain.com"
}

def mock_unsubscribe_script(email: str) -> int:
    """
    Simulates an external script that unsubscribes an email.
    In a real-world scenario, this would be a call to an external service or script.
    Calling a function is much safer than using os.system() or subprocess.run(),
    which can be vulnerable to command injection if not handled properly.

    Args:
        email: The email to unsubscribe.

    Returns:
        0 for success, 1 for failure.
    """
    # This is a mock implementation.
    # It succeeds if the email contains "success" or "test", and fails otherwise.
    if email and ("success" in email or "test" in email):
        print(f"-> Mock script: Successfully unsubscribed {email}")
        return 0  # Success
    else:
        print(f"-> Mock script: Failed to unsubscribe {email}")
        return 1  # Failure

def unsubscribe_user(email: str) -> int:
    """
    Checks if an email exists and attempts to unsubscribe it.

    Args:
        email: The email address to check and unsubscribe.

    Returns:
        1 if the email exists and the unsubscribe script succeeds, otherwise 0.
    """
    # Input validation: ensure email is not None or empty.
    if not isinstance(email, str) or not email.strip():
        print(f"Processing '{email}': Invalid input.")
        return 0
    
    print(f"Processing '{email}':")

    # Step 1: Check if the email exists in the database.
    if email in email_database:
        print("-> Email found in the database.")
        # Step 2: If it exists, call the unsubscribe script.
        script_result = mock_unsubscribe_script(email)

        # Step 3: Check the script's return value.
        if script_result == 0:
            # Script succeeded, return 1.
            print("-> Result: 1 (Success)")
            return 1
        else:
            # Script failed, return 0.
            print("-> Result: 0 (Script Failed)")
            return 0
    else:
        # Email does not exist, return 0.
        print("-> Email not found in the database.")
        print("-> Result: 0")
        return 0

# Main function to run test cases
if __name__ == "__main__":
    test_emails = [
        "user.success@example.com",  # Test case 1: Email exists, script succeeds
        "user.fail@example.com",     # Test case 2: Email exists, script fails
        "not.found@example.com",     # Test case 3: Email does not exist
        "test.user@domain.com",      # Test case 4: Another success case
        None                         # Test case 5: Invalid None input
    ]

    for i, email in enumerate(test_emails, 1):
        print(f"\n--- Test Case {i} ---")
        result = unsubscribe_user(email)
        print(f"Final return value: {result}")